sdk/core/Azure.Core/src/Internal/RehydrationOperation.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal class RehydrationOperation : Operation
+    {
+        private readonly NextLinkOperationImplementation _nextLinkOperation;
+        private readonly OperationInternal _operation;
+
+        public RehydrationOperation(HttpPipeline pipeline, RehydrationToken? rehydrationToken, ClientOptions options = null)
+        {
+            Argument.AssertNotNull(pipeline, nameof(pipeline));
+            Argument.AssertNotNull(rehydrationToken, nameof(rehydrationToken));
+            _nextLinkOperation = (NextLinkOperationImplementation)NextLinkOperationImplementation.Create(pipeline, rehydrationToken);
+            var operationState = _nextLinkOperation.UpdateStateAsync(false, default).EnsureCompleted();
+            _operation = operationState.HasCompleted
+                ? _operation = new OperationInternal(operationState)
+                : new OperationInternal(_nextLinkOperation, new ClientDiagnostics(options ?? ClientOptions.Default), operationState.RawResponse);
+        }
+
+        public override string Id => _nextLinkOperation?.OperationId ?? null;
```

Comment:
```suggestion
        public override string? Id => _nextLinkOperation?.OperationId ?? null;
```

Reply:
Ah, I see that is part of the public API and can't be changed.  When would this return null?  This would be a violation of the public contract (Operation.Id is not nullable), so if we can avoid it, we should.  Can it ever actually happen?

Reply:
removed nullable